### ISSUE TITLE
[experimental codegen] Only hermes-engine pod outdated per RN upg

### ISSUE THREAD

**Issue by leotm** [leotm/react-native-template-new-architecture#825]
tl;dr `pod install` fails each RN upg from OOD Hermes

---

Example
- https://github.com/leotm/react-native-template-new-architecture/pull/815 0.70.0-rc.1 -> 0.70.0-rc.2
- https://github.com/leotm/react-native-template-new-architecture/runs/7703059294?check_suite_focus=true

```console
# ...
[Codegen] Done.
Analyzing dependencies
Fetching podspec for `DoubleConversion` from `../node_modules/react-native/third-party-podspecs/DoubleConversion.podspec`
Fetching podspec for `RCT-Folly` from `../node_modules/react-native/third-party-podspecs/RCT-Folly.podspec`
Fetching podspec for `boost` from `../node_modules/react-native/third-party-podspecs/boost.podspec`
Fetching podspec for `glog` from `../node_modules/react-native/third-party-podspecs/glog.podspec`
Fetching podspec for `hermes-engine` from `../node_modules/react-native/sdks/hermes/hermes-engine.podspec`

Adding spec repo `trunk` with CDN `[https://cdn.cocoapods.org/`](https://cdn.cocoapods.org/%60)
[!] [Codegen] warn: using experimental new codegen integration
[!] CocoaPods could not find compatible versions for pod "hermes-engine":
  In snapshot (Podfile.lock):

    hermes-engine (from `../node_modules/react-native/sdks/hermes/hermes-engine.podspec`)
[!] [Deprecated] You are using the old `libraries` array to list all your codegen.\nThis method will be removed in the future.\nUpdate your `package.json` with a single object.

  In Podfile:
    hermes-engine (from `../node_modules/react-native/sdks/hermes/hermes-engine.podspec`)

It seems like you've changed the version of the dependency `hermes-engine` and it differs from the version stored in `Pods/Local Podspecs`.
You should run `pod update hermes-engine --no-repo-update` to apply changes made locally.
Error: Process completed with exit code 1.
```

`pod update hermes-engine --no-repo-update` shouldn't be required each time, jus like other pods don't

pos regression from bundled hermes?
- https://reactnative.dev/blog/2022/07/08/hermes-as-the-default
- https://github.com/facebook/hermes

will this fix? or is this where to fix itit via Yarn v3 patch?
- https://github.com/leotm/react-native-template-new-architecture/pull/797
- https://app.renovatebot.com/package-diff?name=react-native-codegen&from=0.70.3&to=0.71.0

---

Happening again
- https://github.com/leotm/react-native-template-new-architecture/pull/863

`pod install` forgets to update `- hermes-engine (0.70.0-rc.2)` to `- hermes-engine (0.70.0-rc.3)`
